### ISSUE TITLE
Error in llm chat due to invalid key \e[d from case conversion in key bindings (Windows 11)

### ISSUE THREAD

**Issue by gerardantoun** [simonw/llm#585]
## Description:
 
I'm encountering an error when running the llm chat command on Windows. The error appears to be related to key bindings in the pyreadline3 library, where key descriptions are converted to lowercase, causing certain keys to be unrecognized.

### Error Message:

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\username\AppData\Local\Programs\Python\Python312\Scripts\llm.exe\__main__.py", line 7, in <module>
  File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\click\core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\click\core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\click\core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\click\core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\click\core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\llm\cli.py", line 346, in chat
    readline.parse_and_bind("\\e[D: backward-char")
  File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\pyreadline3\rlmain.py", line 114, in parse_and_bind
    self.mode._bind_key(key, func)
  File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\pyreadline3\modes\basemode.py", line 176, in _bind_key
    keyinfo = make_KeyPress_from_keydescr(key.lower()).tuple()
  File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\pyreadline3\keysyms\common.py", line 191, in make_KeyPress_from_keydescr
    raise IndexError("Not a valid key: '%s'" % keydescr)
IndexError: Not a valid key: '\e[d'
```

## Steps to Reproduce:

Install the llm package on a Windows machine.
Run the command:
```
llm chat -m gpt-4o
```
## Analysis:

The error seems to originate from the llm package attempting to bind key sequences using readline.parse_and_bind. Specifically, in llm/cli.py:

```python
readline.parse_and_bind("\\e[D: backward-char")
```

In the pyreadline3 library used on Windows, the _bind_key method in basemode.py converts the key description to lowercase:

```python
Copy code
keyinfo = make_KeyPress_from_keydescr(key.lower()).tuple()
```

This conversion changes '\e[D' to '\e[d', which is not recognized by pyreadline3, leading to the IndexError.

## Environment:

OS: Windows 11
Python Version: 3.12.7
llm Version: 0.16
pyreadline3 Version: 3.5.4

## Additional Info

The issue seems specific to Windows due to differences in how key inputs are handled compared to Unix-like systems according to chatgpt.

It also suggested to use `prompt_toolkit` instead of `pyreadline3` but it might be a huge change.

**Comment by grota** [comment]
related to https://github.com/simonw/llm/issues/582#issuecomment-2422632951 which in turn is related to a bunch of other issue.
TL;DR: it seems to me as if readline has been introduced first a while ago to fix a problem on mac. Currently it breaks `llm chat` usage on linux and windows.

**Comment by gerardantoun** [comment]
Might be. The issue you mentioned (https://github.com/simonw/llm/issues/582#issuecomment-2422632951) seems to come after the user was able to run `llm chat -m <some model>` successfully.

I am not even getting to start the chat with the LLM. The error happens from writing `llm chat -m gpt-4o`

**Comment by gerardantoun** [comment]
I created a virtual environment, and made the following changes in `.venv\Lib\site-packages\llm\cli.py` as @grota suggested:

```python
# import readline

# readline.parse_and_bind("\\e[D: backward-char")
# readline.parse_and_bind("\\e[C: forward-char")
```

It worked, I get to the chat part in this case (which I wasn't able to get to before):

![image](https://github.com/user-attachments/assets/b8f895f1-c893-4489-ba37-0f56997974cf)
